sdk/state/ingest.go

```diff
@@ -15,39 +15,37 @@ import (
 // determine if successful or not, and understand changes in the ledger as a
 // result.
 func (c *Channel) IngestTx(tx *txnbuild.Transaction) error {
-	// Use the tx to update the latest unauthorized close agreement if possible.
+	// TODO: Use the transaction result to affect on success/failure.
+
+	c.ingestTxToUpdateInitiatorEscrowAccountSequence(tx)
+
 	err := c.ingestTxToUpdateUnauthorizedCloseAgreement(tx)
 	if err != nil {
 		return err
 	}
 
-	// TODO: If the tx hash matches an authorized or unauthorized declaration,
-	// mark the channel as closing.
-	// TODO: If the tx hash matches an authorized or unauthorized close, mark the
-	// channel as closed.
-	// TODO: If the tx is for an older declaration, mark the channel as closing with
-	// requiring bump.
-	// TODO: Use the transaction result to affect on success/failure.
-
 	return nil
 }
 
-// ingestTxToUpdateUnauthorizedCloseAgreement uses the signatures in the
-// transaction to authorize an unauthorized close agreement if the channel has
-// one. This process helps to give a participant who proposed an agreement the
+func (c *Channel) ingestTxToUpdateInitiatorEscrowAccountSequence(tx *txnbuild.Transaction) {
+	// If the transaction's source account is not the initiator's escrow
+	// account, then the transaction is not a part of a close agreement.
+	if tx.SourceAccount().AccountID != c.initiatorEscrowAccount().Address.Address() {
+		return
+	}
+
+	c.setInitiatorEscrowAccountSequence(tx.SourceAccount().Sequence)
+}
+
+// ingestTxToUpdateUnauthorizedCloseAgreement uses the signatures in the transaction to
+// authorize an unauthorized close agreement if the channel has one.
+// This process helps to give a participant who proposed an agreement the
 // ability to close the channel if they did not receive the confirmers
 // signatures for a close agreement when the agreement was being negotiated. If
 // the transaction cannot be used to do this the function returns a nil error.
 // If the transaction should be able to provide this data and cannot, the
 // function errors.
 func (c *Channel) ingestTxToUpdateUnauthorizedCloseAgreement(tx *txnbuild.Transaction) error {
-	// If the transaction's source account is not the initiator's escrow
-	// account, then the transaction will not be a transaction for a close
-	// agreement and won't be able to update any unauthorized close agreement.
-	if tx.SourceAccount().AccountID != c.initiatorEscrowAccount().Address.Address() {
-		return nil
-	}
-
```

Comment:
I think you need to add this back. This function should ignore any tx that isn't for the EI account. (All the changes to this function look superficial and unrelated to setting the channel closing state.)

Reply:
got lost in the shuffle, added back here: https://github.com/stellar/experimental-payment-channels/pull/232/commits/79e69df149389d443cd0332ac45329c29e1857bb

interestingly we actually don't need it since the tx hashes are being compared. But I agree to end early if it's not the EI account